src/coreclr/hosts/unixcoreruncommon/CMakeLists.txt

```diff
@@ -2,6 +2,8 @@ project(unixcoreruncommon)
 
 add_compile_options(-fPIC)
 
+check_symbol_exists(getauxval sys/auxv.h HAVE_GETAUXVAL)
```

Comment:
Does it show line with `HAVE_GETAUXVAL - Success` during the initial cmake configuration run?
I think maybe it only requires `check_function_exists` and can go in a separate file?
`cmakedefine` or `cmakedefine01` would also be needed in configuration file.

`config.h.in`:

```h
#cmakedefine01 HAVE_GETAUXVAL
```

then new file in same directory: `configure.cmake`

```cmake
include(CheckFunctionExists)

check_function_exists(getauxval HAVE_GETAUXVAL)

configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
```

and this CMakeLists.txt to have `include(configure.cmake)`.

Then in `src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp`:

```diff
- #if defined(HAVE_GETAUXVAL)
+ #include "config.h"
+ #if HAVE_GETAUXVAL
```

Reply:
I would also prefer following the way we use at other places as @am11 suggested (adding config.h.in and configure.cmake) for the sake of consistency.

Reply:
OK, will do this.